url/join_test.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "../assert/mod.ts";
+import * as url from "./mod.ts";
+
+const TESTSUITE: [[string | URL, ...string[]], URL][] = [
+  [
+    ["https://deno.land", "std", "assert", "mod.ts"],
+    new URL("https://deno.land/std/assert/mod.ts"),
+  ],
+  [
+    [new URL("https://deno.land"), "std", "assert", "mod.ts"],
+    new URL("https://deno.land/std/assert/mod.ts"),
+  ],
+  [
+    [new URL("https://deno.land//"), "/std/", "/assert/", "//mod.ts"],
+    new URL("https://deno.land/std/assert/mod.ts"),
+  ],
+  [["https://deno.land/"], new URL("https://deno.land/")],
+];
+
+Deno.test("join", function () {
+  for (const [[test_url, ...paths], expected] of TESTSUITE) {
+    assertEquals(url.join(test_url, ...paths), expected);
+  }
+});
```

Comment:
Now that I'm looking at the tests, I'm not sure if join even makes sense.

Reply:
Why is it so?

Reply:
I guess it handles edgecases around double `/` so maybe it's still useful. Not entirely sure what the utility is. Thoughts @marvinhagemeister?